aws/ami/files/scylla_install_ami

```diff
@@ -60,64 +68,106 @@ if __name__ == '__main__':
     os.remove('/var/lib/scylla-housekeeping/housekeeping.uuid')
 
-    os.remove('/etc/udev/rules.d/80-net-name-slot.rules')
+    if distro == 'centos':
+        os.remove('/etc/udev/rules.d/80-net-name-slot.rules')
     with open('/etc/default/grub') as f:
         grub = f.read()
     grub = re.sub(r'^GRUB_CMDLINE_LINUX="(.+)"$', r'GRUB_CMDLINE_LINUX="\1 net.ifnames=0"', grub, flags=re.MULTILINE)
     with open('/etc/default/grub', 'w') as f:
         f.write(grub)
-    run('grub2-mkconfig -o /boot/grub2/grub.cfg')
+    if distro == 'centos':
+        run('grub2-mkconfig -o /boot/grub2/grub.cfg')
+    else:
+        run('grub-mkconfig -o /boot/grub/grub.cfg')
 
     with open('/etc/cloud/cloud.cfg') as f:
-        cfg = f.read()
-    cfg2 = re.sub('^ssh_deletekeys:   0', 'ssh_deletekeys:   1', cfg, flags=re.MULTILINE)
-    cfg2 = re.sub('^    name: centos', '    name: scyllaadm', cfg2, flags=re.MULTILINE)
+        lines = f.readlines()
+    s = StringIO()
+    for l in lines:
+        if not re.match(r'^ - mounts\n$', l):
+            s.write(l)
+    cfg = s.getvalue()
+    s.close()
+    if distro == 'centos':
+        cfg = re.sub('^ssh_deletekeys:   0', 'ssh_deletekeys:   1', cfg, flags=re.MULTILINE)
+        cfg = re.sub('^    name: centos', '    name: scyllaadm', cfg, flags=re.MULTILINE)
+    else:
+        cfg = re.sub('^preserve_hostname: false', 'preserve_hostname: false\n\nssh_deletekeys: true', cfg, flags=re.MULTILINE)
+        cfg = re.sub('^     name: ubuntu', '     name: scyllaadm', cfg, flags=re.MULTILINE)
     with open('/etc/cloud/cloud.cfg', 'w') as f:
-        f.write(cfg2)
-    shutil.move('{}/99_user_alias.cfg'.format(homedir), '/etc/cloud/cloud.cfg.d')
-
-    # We need to install LTS kernel for the AMI.
-    # However, elrepo-kernel only provides x86_64 package,
-    # and centos-kernel (altarch/7/kernel) only provides aarch64 package.
-    # So we have to use different kernel here.
-    if platform.machine() == 'x86_64':
-        run('yum -y install grubby')
-        run('rpm --import https://www.elrepo.org/RPM-GPG-KEY-elrepo.org')
-        run('rpm -Uvh http://www.elrepo.org/elrepo-release-7.0-3.el7.elrepo.noarch.rpm')
-        run('yum -y --enablerepo=elrepo-kernel install kernel-ml')
-        mlkver = get_kver('/boot/vmlinuz-*el7.elrepo.x86_64')
-        run('grubby --grub2 --set-default /boot/vmlinuz-{mlkver}'.format(mlkver=mlkver))
-        with open('/etc/yum.conf', 'a') as f:
-            f.write(u'exclude=kernel kernel-devel')
-    elif platform.machine() == 'aarch64':
-        run('yum -y --enablerepo=centos-kernel update kernel')
+        f.write(cfg)
+    run('userdel -f {}'.format(distro))
+    run('cloud-init clean')
+    run('cloud-init init')
+    for skel in glob.glob('/etc/skel/.*'):
+        if not skel.endswith('.bash_profile'):
+            shutil.copy(skel, '/home/scyllaadm')
+    run('useradd -o -u 1000 -g scyllaadm -s /bin/bash -d /home/scyllaadm centos')
```

Comment:
Should this run on CentOS only?

Reply:
Avi requested to create an alias for 'centos' to keep compatibility with existing user scripts:
https://docs.google.com/document/d/1ZuJS37RwkU4JBTC4yykQ2TsXsyiDDkJn3fMILQiDBzI/edit?disco=AAAALNgq6jo

So, on this AMI, default login user is 'scyllaadm', and two aliases 'ubuntu', 'centos'